internal/flux/registry_mirror.go

```diff
@@ -0,0 +1,130 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package flux
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2"
+	"github.com/fluxcd/pkg/apis/kustomize"
+	"gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	greenhousev1alpha1 "github.com/cloudoperators/greenhouse/api/v1alpha1"
+)
+
+const (
+	registryMirrorConfigKey = "containerRegistryConfig"
+)
+
+// RegistryMirrorConfig represents the registry mirror configuration structure.
+type RegistryMirrorConfig struct {
+	PrimaryMirror   string                    `yaml:"primaryMirror"`
+	RegistryMirrors map[string]RegistryMirror `yaml:"registryMirrors"`
+}
+
+// RegistryMirror represents a single registry mirror configuration.
+type RegistryMirror struct {
+	BaseDomain string `yaml:"basedomain"`
+	SubPath    string `yaml:"subPath"`
+}
+
+// GetRegistryMirrorConfig retrieves and parses the registry mirror configuration from the Organization's ConfigMap.
+// Registry mirrors redirect image pulls from original registries to mirror registries.
+// For example, with config:
+//
+//	registryMirrors:
+//	  ghcr.io:
+//	    basedomain: "keppel.eu-de-1.cloud.sap"
+//	    subPath: "ccloud-ghcr-io-mirror"
+//
+// An image like "ghcr.io/cloudoperators/greenhouse:main" becomes "keppel.eu-de-1.cloud.sap/ccloud-ghcr-io-mirror/cloudoperators/greenhouse:main"
```

Comment:
We cannot have SAP internal links as part of the source code. Could you replace this with some generic examples?